### PR TITLE
feat: warning user about manually deploying review apps

### LINKED ISSUE #899: When manually deploying a pull request, explicit that it comes from a fork
When deploying a review app from a PR/MR id, user should be aware if the request is actually coming from a fork of the app.

For now, the API does not give us enough information to know more about the requested PR.

Needs https://github.com/Scalingo/api/issues/2173 in order to be possible

```[tasklist]
### Required by this issue
- [ ] https://github.com/Scalingo/api/issues/2173
```

### PR DESCRIPTION
**Pull request by yohann-bacha**
Fixes #899 

Needs https://github.com/Scalingo/go-scalingo/issues/327 to be done for merging this

- [ ] Add a changelog entry in the section "To Be Released" of CHANGELOG.md

### CODE REVIEW

File: cmd/integration_link.go [hunk 1]
```diff
@@ -414,9 +416,33 @@ List of available integrations:
 
 			utils.CheckForConsent(c.Context, currentApp, utils.ConsentTypeContainers)
 
-			pullRequestID := c.Args().First()
+			pullRequestID, err := strconv.Atoi(c.Args().First())
+			if err != nil {
+				errorQuit(errgo.Notef(err, "invalid pull request id"))
```
Comment (EtienneM) [inline cmd/integration_link.go:421]: question: aren't there some cases where a merge request could contain something else than integer? For instance on GitLab? I'm not completely keen on checking here if it's an integer. Why don't we simply trust the backend which forwards the query to GitHub/GitLab to know if it's valid or not?

suggestion:

```suggestion
				errorQuit(errgo.Notef(err, "invalid pull / merge request id"))
```

File: cmd/integration_link.go [hunk 2]
```diff
@@ -414,9 +416,33 @@ List of available integrations:
...
+
+			pullRequest, err := integrationlink.PullRequest(c.Context, currentApp, pullRequestID)
+			if err != nil {
+				errorQuit(err)
+			}
+
+			if pullRequest.OpenedFromAForkedRepo {
+				awareOfSecurityRisks := c.Bool("aware-of-security-risks")
+				if !awareOfSecurityRisks {
+					io.Info("\nYou are about to deploy a Review App from a Pull Request opened from a fork.")
```
Comment (EtienneM) [inline cmd/integration_link.go:432]: suggestion: Shouldn't we add a message to explain again the risk it implies?

File: cmd/integration_link.go [hunk 3]
```diff
@@ -414,9 +416,33 @@ List of available integrations:
...
+					allowReviewAppsFromForks, err := askForConfirmationToAllowReviewAppsFromForks("Deploy this Pull Request coming from a forked repository ?")
```
Comment (EtienneM) [inline cmd/integration_link.go:433]:
```suggestion
					allowReviewAppsFromForks, err := askForConfirmationToAllowReviewAppsFromForks("Deploy this Pull Request coming from a forked repository?")
```

**Review by EtienneM — CHANGES_REQUESTED** [review]
Needs to be synced with a go-scalingo release

File: cmd/integration_link.go (hunk 2 above)
Comment (yohann-bacha) [reply cmd/integration_link.go:432]: This is done in the askForConfirmation[..] function

File: cmd/integration_link.go (hunk 1 above)
Comment (yohann-bacha) [reply cmd/integration_link.go:421]: Well actually, in the backend, we are speaking about merge/request numbers. Here in the CLI, we are mentioning "id". Should we stick to the "number" denomination, and fix it in the CLI argsusage ?

**Comment by yohann-bacha** [comment]
@EtienneM The changes in go-scalingo hasn't been released yet (but merged on master), what is the procedure already to make the cli build?

File: cmd/integration_link.go (hunk 1 above)
Comment (EtienneM) [reply cmd/integration_link.go:421]: My question is not really a wording question. From what I think, the most important question is: I'm not completely keen on checking here if it's an integer. Why don't we simply trust the backend which forwards the query to GitHub/GitLab to know if it's valid or not?

**Review by EtienneM — COMMENTED** [review]
> @EtienneM The changes in go-scalingo hasn't been released yet (but merged on master), what is the procedure already to make the cli build?

Instructions to release go-scalingo: https://github.com/Scalingo/go-scalingo/#release-a-new-version

File: cmd/integration_link.go (hunk 1 above)
Comment (yohann-bacha) [reply cmd/integration_link.go:421]: Oh, your comment was on the error message line, I thought you were talking about it. 

Yes, we can rely on the backend for the args coherence

Comment (yohann-bacha) [reply cmd/integration_link.go:421]: @EtienneM Hm, well actually go-scalingo takes a number, and have just been released. To take a string, we should make a 6.6.1. Is it worth? :confused:

Comment (EtienneM) [reply cmd/integration_link.go]: Groumpf.... OK :(

**Review by EtienneM — APPROVED** [review]

**Pull request merged**